solutions/instances/main.tf

```diff
@@ -10,6 +10,8 @@ locals {
   validate_existing_cloud_monitoring = var.cloud_monitoring_provision && var.existing_cloud_monitoring_crn != null ? tobool("if cloud_monitoring_provision is set to true, then existing_cloud_monitoring_crn should be null and vice versa") : true
   # tflint-ignore: terraform_unused_declarations
   validate_cos_resource_group = var.existing_cos_instance_crn == null ? var.ibmcloud_cos_api_key != null && var.cos_resource_group_name == null ? tobool("if value for `ibmcloud_cos_api_key` is set, then `cos_resource_group_name` cannot be null") : true : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_metric_routing_creation = var.enable_metric_routing_to_cloud_monitoring ? ((var.existing_cloud_monitoring_crn != "" || var.cloud_monitoring_provision) ? true : tobool("When `enable_metric_routing_to_cloud_monitoring` is set to true, you must either set `cloud_monitoring_provision` as true or provide the `existing_cloud_monitoring_crn`.")) : true
```

Comment:
We use `null` always instead of `""`
So condition needs a change `var.existing_cloud_monitoring_crn != ""`